src/classes/CON_DoNotContact_TDTM.cls

```diff
@@ -32,95 +32,177 @@
-                else if (con.Do_Not_Contact__c != oldCon.Do_Not_Contact__c)
-                    markContactDoNotContact(con);
-            }
+        if (this.isBeforeUpdate(triggerAction)) {
+            Map<Id, Contact> oldUpdatedContactsById = new Map<Id, Contact>((List<Contact>)oldlist);
+            this.processBeforeUpdate(newContactsList, oldUpdatedContactsById);
         }
 
         return new DmlWrapper();
-	}
+    }
+    
+    /**
+     * @description Determine if trigger fired is the result of a BeforeInsert action. 
+     * @param triggerAction The current trigger action
+     * @return returns true is trigger context is BeforeInsert, false otherwise
+     */ 
+    @TestVisible
+    private Boolean isBeforeInsert(TDTM_Runnable.Action triggerAction) {
+        return triggerAction == TDTM_Runnable.Action.BeforeInsert;
+    }
 
-    /*******************************************************************************************************
-    * @description Toggles additional Contact fields as necessary when a contact is flagged as deceased. 
-    * @param con The Contact record.
-    * @return void.  
-    ********************************************************************************************************/
-    public void markContactDeceased(Contact con) {
+    /**
+     * @description Determine if trigger fired is the result of a BeforeUpdate action. 
+     * @param triggerAction The current trigger action
+     * @return returns true if trigger context is BeforeUpdate, false otherwise
+     */ 
+    @TestVisible
+    private Boolean isBeforeUpdate(TDTM_Runnable.Action triggerAction) {
+        return triggerAction == TDTM_Runnable.Action.BeforeUpdate;
+    }
 
-        con.Do_Not_Contact__c = con.Deceased__c;
-        markContactDoNotContact(con);
+    /**
+     * @description Execute the logic necessary for a BeforeInsert trigger action.
+     * @param newContactsList List of Contacts inserted
+     */ 
+    @TestVisible
+    private void processBeforeInsert(List<Contact> newContactsList) {
+        List<Contact> deceasedContactsList = new List<Contact>();
+        List<Contact> doNotContactContactsList = new List<Contact>();
+
+        for (Contact con : newContactsList){
+            if (con.Deceased__c == true){
+                con.Do_Not_Contact__c = con.Deceased__c;
+                deceasedContactsList.add(con);
+            }
 
-        con.Exclude_from_Household_Name__c = con.Deceased__c;
-        con.Exclude_from_Household_Formal_Greeting__c = con.Deceased__c;
-        con.Exclude_from_Household_Informal_Greeting__c = con.Deceased__c;
+            if (con.Do_Not_Contact__c == true){
+                doNotContactContactsList.add(con);
+            }
+        }
 
-        copyNamingExclusionsFromCheckboxes(con);
+        if (deceasedContactsList.isEmpty() == false){
+            this.syncNamingExclusionsOnContacts(deceasedContactsList);
+            this.syncDoNotContactFieldsOnContacts(deceasedContactsList);
```

Comment:
@m-kinkade is this line necessary? Do Not Contact field is getting set on line 97, meaning this contact will also get added to the doNotContactContactsList. We're running the same contact through the syncDoNotContactFieldsOnContacts method twice.

Reply:
Great catch, I've updated and rerun the tests to ensure nothing broke in the process.